packages/atomic/src/components/commerce/atomic-commerce-interface/atomic-commerce-interface.tsx

```diff
@@ -0,0 +1,480 @@
+   * @example:
+   * ```html
+   * <script nonce="<YOUR_GENERATED_NONCE>">
+   *  import {setNonce} from '@coveo/atomic';
+   *  setNonce('<YOUR_GENERATED_NONCE>');
+   * </script>
+   * ```
+   */
+  @Prop({reflect: true}) public CspNonce?: string;
+
+  /**
+   * A reference clone of the interface i18next instance.
+   */
+  private i18nClone!: i18n;
+
+  public constructor() {
+    this.commonInterfaceHelper = new CommonAtomicInterfaceHelper(
+      this,
+      'CoveoAtomic'
+    );
+  }
+
+  public connectedCallback() {
+    this.store.setLoadingFlag(FirstSearchExecutedFlag);
+    this.i18nClone = this.i18n.cloneInstance();
+    this.i18n.addResourceBundle = (
+      lng: string,
+      ns: string,
+      resources: object,
+      deep?: boolean,
+      overwrite?: boolean
+    ) => this.addResourceBundleWithWarning(lng, ns, resources, deep, overwrite);
+  }
+
+  componentWillLoad() {
+    if (this.CspNonce) {
+      setNonce(this.CspNonce);
+    }
+    this.initAriaLive();
+    this.initFieldsToInclude();
+  }
+
+  @Watch('analytics')
+  public toggleAnalytics() {
+    if (!this.commonInterfaceHelper.engineIsCreated(this.engine)) {
+      return;
+    }
+
+    this.commonInterfaceHelper.onAnalyticsChange();
+  }
+
+  @Watch('language')
+  public updateLanguage() {
+    if (!this.commonInterfaceHelper.engineIsCreated(this.engine)) {
+      return;
+    }
+    this.commonInterfaceHelper.onLanguageChange();
+  }
+
+  @Watch('iconAssetsPath')
+  public updateIconAssetsPath() {
+    this.store.set('iconAssetsPath', this.iconAssetsPath);
+  }
+
+  public disconnectedCallback() {
+    this.unsubscribeUrlManager();
+    this.unsubscribeSearchStatus();
+    window.removeEventListener('hashchange', this.onHashChange);
+  }
+
+  @Listen('atomic/initializeComponent')
+  public handleInitialization(event: InitializeEvent) {
+    this.commonInterfaceHelper.onComponentInitializing(event);
+  }
+
+  @Listen('atomic/scrollToTop')
+  public scrollToTop() {
+    const scrollContainerElement = document.querySelector(this.scrollContainer);
+    if (!scrollContainerElement) {
+      this.bindings.engine.logger.warn(
+        `Could not find the scroll container with the selector "${this.scrollContainer}". This will prevent UX interactions that require a scroll from working correctly. Please check the CSS selector in the scrollContainer option`
+      );
+      return;
+    }
+
+    scrollContainerElement.scrollIntoView({behavior: 'smooth'});
+  }
+
+  /**
+   * Initializes the connection with the headless search engine using options for accessToken (required), organizationId (required), renewAccessToken, organizationEndpoints (recommended), and platformUrl (deprecated).
+   */
+  @Method() public initialize(options: CommerceInitializationOptions) {
+    return this.internalInitialization(() => this.initEngine(options));
+  }
+
+  @Method() public initializeWithEngine(engine: CommerceEngine) {
```

Comment:
We should not need both `initializeWithEngine` and `initializeWithCommerceEngine`. It's the same function